docs/migration/cosmoshub-4-v9-Lambda-upgrade.md

```diff
@@ -0,0 +1,297 @@
+```
+
+Stop the node, and switch the binary to Gaia v9.0.0 and re-start by `gaiad start`.
+
+It may take several minutes to a few hours until validators with a total sum voting power > 2/3 to complete their nodes upgrades. After that, the chain can continue to produce blocks.
+
+### Method II: Upgrade using Cosmovisor
+
+> **Warning**  <span style="color:red">**Please Read Before Proceeding**</span><br>
+> **Using Cosmovisor 1.2.0 and higher requires a lowercase naming convention for upgrade version directory. For Cosmovisor 1.1.0 and earlier, the upgrade version is not lowercased.**
+>
+> **For Example:** <br>
+> **Cosmovisor =< `1.1.0`: `/upgrades/v9-Lambda/bin/gaiad`**<br>
+> **Cosmovisor >= `1.2.0`: `/upgrades/v9-lambda/bin/gaiad`**<br>
+
+| Cosmovisor Version | Binary Name in Path |
+|--------------------|---------------------|
+| 1.3                | v9-lambda           |
+| 1.2                | v9-lambda           |
+| 1.1                | v9-Lambda           |
+| 1.0                | v9-Lambda           |
+
+
+
+### _Manually preparing the Gaia v9.0.0 binary_
+
+##### Preparation
+
+Install the latest version of Cosmovisor (`1.3.0`):
+
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
+```
+
+**Verify Cosmovisor Version**
+```shell
+cosmovisor version
+cosmovisor version: v1.3.0
+```
+
+Create a cosmovisor folder:
+
+create a Cosmovisor folder inside `$GAIA_HOME` and move Gaia v8.0.1 into `$GAIA_HOME/cosmovisor/genesis/bin`
+
+```shell
+mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+````
+
+build Gaia v9.0.0, and move gaiad v9.0.0 to `$GAIA_HOME/cosmovisor/upgrades/v9-lambda/bin`
+
+```shell
+mkdir -p  $GAIA_HOME/cosmovisor/upgrades/v9-lambda/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/upgrades/v9-lambda/bin
+```
+
+Then you should get the following structure:
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── gaiad  #v8.0.1
+└── upgrades
+    └── v9-lambda
+        └── bin
+            └── gaiad  #v9.0.0
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+export DAEMON_HOME= $GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Start the node:
+
+```shell
+cosmovisor start --x-crisis-skip-assert-invariants
+```
+
+Skipping the invariant checks is strongly encouraged since it decreases the upgrade time significantly and since there are some other improvements coming to the crisis module in the next release of the Cosmos SDK.
+
+#### Expected upgrade result
+
+When the upgrade block height is reached, Gaia will panic and stop:
+
+This may take 7 minutes to a few hours.
+After upgrade, the chain will continue to produce blocks when validators with a total sum voting power > 2/3 complete their nodes upgrades.
+
+### _Auto-Downloading the Gaia v9.0.0 binary (not recommended!)_
```

Comment:
Why do we even have this if it's not recommended?

Reply:
It's still an available method

Reply:
this is more like auto-download is easy to use, but users have to keep in mind the verification of binary of download is automatic, if they want to be very secure, they should verify the download themselves as well

Reply:
Resolving based on available method